### DISCUSSION TITLE
1.0-pre3: Windows fix

### DISCUSSION THREAD

**Discussion by Dadoum** [Dadoum/Sideloader#31] [Announcements]
Mainly includes a fix for the Windows CLI tool. Also adds the missing button on some distros to download the requirements.

<hr /><em>This discussion was created from the release <a href='https://github.com/Dadoum/Sideloader/releases/tag/1.0-pre3'>1.0-pre3: Windows fix</a>.</em>

**Comment by dk865** [comment]
I'm just wondering, can we can just unzip the file, then run it?

If that's how it's supposed to work on Windows (11), then it doesn't work for me. I read through the README but it seemed complex, and I don't have admin. I do have admin on a Windows 10, 2011 Sony computer, but it takes 60+ Minutes to turn on, and 10+ minutes to open file explorer, which usually doesn't work very good, and freezes + goes unresponsive if you right click, or triple click. Which, I would be okay using that, but it has sideloadly, and I want a sideloader for my Non-Admin Windows 11 PC which is 500% Faster, I try and keep the old PC for only jailbreaking, and necessary stuff, because of its slowness.

Now, if anyone wants my terminal messages, I will send them, it's just I was wondering if I am doing something wrong out of the box.

I run, `.\sideloader-cli-windows-x86_64.exe install "F:\JB\IPA's\Dopamine.ipa" -d` to try and sideload Dopamine as 1 of the 6 different apps I tested, and have tried on 5 different devices, 2 different iPod Touch 5th Gens, iOS 9.3.5, Phoenix, with AppSync Unified Installed, an iPhone 13 Mini, iOS 17.5 Developer Beta, iPad Air 2, iPadOS 15.8 (meowbrek2 used to be installed, but I got rid of it to try to switch to nekoJB rootful a few days ago, which didn't work lol, and I already uninstalled the rootles) (which leads me to assume it's a computer error) and lastly an iPod Touch 4th Generation, iOS 6.1.6, p0sixpwn, and AppSync Unified.

Does anyone have any ideas?

Thanks.

**Reply by Dadoum** [reply]
No it isn't, I should bundle few binaries with it but I haven't got around adding that in the automated tool, so here they are: 
[libimobiledevice.1.2.1-r1122-win-x64.zip](https://github.com/Dadoum/Sideloader/files/14964898/libimobiledevice.1.2.1-r1122-win-x64.zip)

Sideloader still requires Apple's Mobile device driver, which is bundled with iTunes unfortunately too. You can still try but it will probably not detect your device without it.

**Reply by dk865** [reply]
Thank you for the response!

after unzipping the contents of libimobiledevice, I moved it all to the same folder of the executable, and tried to run it. When I did with any arguments, nothing happened, it just wait a few seconds, then terminated, with no text at all! This is what it looks loke:

PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64> .\sideloader-cli-windows-x86_64.exe -h
PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64> .\sideloader-cli-windows-x86_64.exe install "F:\JB\IPA's\TrollInstallerX.ipa"
PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64> .\sideloader-cli-windows-x86_64.exe install "F:\JB\IPA's\TrollInstallerX.ipa" -d
PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64>

And, when I moved them to a different folder and added them to my personal path using run `rundll32 sysdm.cpl,EditEnvironmentVariables` it had this error:

PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64> .\sideloader-cli-windows-x86_64.exe -h
Usage: sideloader-cli-windows-x86_64.exe [-d] [-h] <command> [<args>]

Available commands:
  app-id         Manage App IDs.
  cert           Manage certificates.
  install        Install an application on the device (renames the app, register
                 the identifier, sign and install automatically).
  sign           Sign an application bundle.
  team           Manage teams.
  tool           Run Sideloader's tools.
  version        Print the version.

Optional arguments:
  -d, --debug    Enable debug logging
  -h, --help     Show this help message and exit

PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64> .\sideloader-cli-windows-x86_64.exe install "F:\JB\IPA's\TrollInstallerX.ipa" -d
            cli_frontend ERROR 2024-04-13T07:58:59.855 std.zlib.ZlibException at std\zlib.d:98: data error
            cli_frontend DEBUG 2024-04-13T07:58:59.902 Full exception: std.zlib.ZlibException@std\zlib.d(98): data error
----------------
0x00007FF7ED448593
0x00007FF7ED4482C4
0x00007FF7ED3582ED
0x00007FF7ED42C4A1
0x00007FF7ED326785
0x00007FF7ED44C403
0x00007FF7ED368223
0x00007FF7ED2AFE71
0x00007FF7ED2B00C7
0x00007FF7ED3A1B61
0x00007FF7ED3A1D1B
0x00007FF7ED403918
0x00007FF83A079363 in recalloc
0x00007FF83BA1257D in BaseThreadInitThunk
0x00007FF83CD6AA58 in RtlUserThreadStart
PS C:\Users\st29dkolb\Downloads\sideloader-cli-windows-x86_64>

At least the help command works, but nothing else does. I think it's just not really in the path there, but I'm confused why it doesn't work when it's in the same folder... Any other ideas?

**Comment by TexasBrandon** [comment]
Thanks for making this Linux build.  Sideloadly is Windows / Mac only and I refuse to use Windows and don't have a macbook on hand.  Made this simple.

**Comment by dk865** [comment]
Now that I'm on Linux, it works wonderfully! The GUI is nice on my end too. Very nice job! (: Also, do you believe you will be rewriting iCode? That would be nice, or if you could make something Linux wise to turn a Cordova project, or an xcodeproj file into and Ipa file!

**Comment by Milli189** [comment]
Thank you so much for this! I've starred the project!
I'm on Linux and I really needed an app like that, on windows I've used sideloadly but it was super buggy, with this instead I have nice looking app that is easy and intuitive to use!
Thank you so much for this!
